instana/restapi/synthetic-monitor-rest-resource.go

```diff
@@ -0,0 +1,72 @@
+package restapi
+
+import "errors"
+
+// NewSyntheticMonitorRestResource creates a new REST resource using the provided unmarshaller function to convert the response from the REST API to the corresponding InstanaDataObject. The REST resource is using PUT as operation for create and update
+func NewSyntheticMonitorRestResource(unmarshaller JSONUnmarshaller, client RestClient) RestResource {
```

Comment:
Is this really needed? I think the generic REST resource would work here as well

Reply:
I started out without. However the PUT returns an empty response. And the generic REST resource expects the full body to be returned on a PUT. So I had to change the logic a bit for the PUT request (happens when the object is changed).